academic_observatory_workflows/workflows/openalex_telescope.py

```diff
@@ -356,53 +363,70 @@ def __init__(
+            task_branch = BranchPythonOperator(
+                task_id='branch',
+                python_callable=partial(self.task_callable, self.determine_branches),
             )
-        )
+
+            task_groups = []
+            for entity_name in self.entity_names:
+                with TaskGroup(group_id=entity_name) as tg:
+                    task_snapshot = self.make_python_operator(self.bq_snapshot, "bq_snapshot", op_kwargs={"entity_name": entity_name})
+                    task_download = make_download_bash_op(self, entity_name, "download", trigger_rule=TriggerRule.NONE_FAILED)
+                    task_transform = self.make_python_operator(self.transform, "transform", op_kwargs={"entity_name": entity_name})
+                    task_upload = self.make_python_operator(self.upload_upsert_files, "upload", op_kwargs={"entity_name": entity_name})
+                    task_bq_load_upserts = self.make_python_operator(self.bq_load_upsert_tables, "bq_load_upserts", op_kwargs={"entity_name": entity_name})
+                    task_bq_upsert_records = self.make_python_operator(self.bq_upsert_records, "bq_upsert_records", op_kwargs={"entity_name": entity_name})
+                    task_bq_load_deletes = self.make_python_operator(self.bq_load_delete_tables, "bq_load_deletes", op_kwargs={"entity_name": entity_name})  # This task can skip
+                    task_bq_delete_records = self.make_python_operator(self.bq_delete_records, "bq_delete_records", op_kwargs={"entity_name": entity_name}, trigger_rule=TriggerRule.NONE_FAILED)  # This task can skip
+                    task_add_dataset_releases = self.make_python_operator(self.add_dataset_releases, "add_dataset_releases", op_kwargs={"entity_name": entity_name}, trigger_rule=TriggerRule.NONE_FAILED)
+
+                    (
+                        task_snapshot
+                        >> task_download
+                        >> task_transform
+                        >> task_upload
+                        >> task_bq_load_upserts
+                        >> task_bq_upsert_records
+                        >> task_bq_load_deletes
+                        >> task_bq_delete_records
+                        >> task_add_dataset_releases
+                    )
+                    task_groups.append(tg)
+
+            task_cleanup = self.make_python_operator(self.cleanup, "cleanup", trigger_rule=TriggerRule.NONE_FAILED)
+            task_wait = EmptyOperator(task_id=external_task_id)
+
+            # Link tasks together
+            (
+                    task_sensor
+                    >> task_check_dependencies
+                    >> task_fetch_releases
+                    >> task_create_datasets
+                    >> task_aws_to_gcs_transfer
+                    >> task_branch
+                    >> task_groups
+                    >> task_cleanup
+                    >> task_wait
+            )
+            # fmt: on
```

Comment:
Does black not like the airflow task linking?

Reply:
I found it tidier by having each task definition on a single line, although I could tighten up the part that isn't formatted.